openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -19,42 +19,52 @@
             except for adding ancestry tags to the taxa table.
 """
 
+import logging
 import os
+import zipfile
+from datetime import timedelta
 from pathlib import Path
 
 import pendulum
-from airflow.exceptions import AirflowSkipException
+import requests
+from airflow.exceptions import AirflowNotFoundException, AirflowSkipException
 from airflow.operators.python import PythonOperator
 from airflow.providers.amazon.aws.hooks.s3 import S3Hook
 from airflow.providers.postgres.hooks.postgres import PostgresHook
 from airflow.providers.postgres.operators.postgres import PostgresOperator
 from airflow.utils.task_group import TaskGroup
-from common.constants import POSTGRES_CONN_ID
-from common.licenses import NO_LICENSE_FOUND, get_license_info
-from common.loader import provider_details as prov
+from airflow.utils.trigger_rule import TriggerRule
+from common.constants import POSTGRES_CONN_ID, XCOM_PULL_TEMPLATE
+from common.loader import provider_details, reporting, sql
 from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
 
 
+logger = logging.getLogger(__name__)
+
 AWS_CONN_ID = os.getenv("AWS_CONN_ID", "test_conn_id")
-PROVIDER = prov.INATURALIST_DEFAULT_PROVIDER
+OPENVERSE_BUCKET = os.getenv("OPENVERSE_BUCKET", "openverse-storage")
```

Comment:
Great question! There is a distinction between an _environment variable_ and an _Airflow Variable_ that might be a bit difficult to suss out. We can define Airflow Variables using env vars, but Airflow Variables can also be defined in the UI and overwritten/changed at any time. Additionally, storing secrets (e.g. API keys) in Airflow Variables will [automatically mask them in the logs](https://airflow.apache.org/docs/apache-airflow/stable/security/secrets/mask-sensitive-values.html). Lastly, we can use the `deserialize_json` flag to convert Airflow Variable values to native python types, which was something I mentioned on #904. Airflow Variables can be changed at any time whereas environment variable changes require a deployment (or at least a service restart). There's some more context for this in https://github.com/WordPress/openverse-catalog/issues/209.

> I recently learned in #904 that we can use template strings for a lot of environment variables.

To be clear, the templating that I described in #904 only applies to Airflow Variables, not environment variables. The former we wished to defer because they issued a database query on ever `Variable.get`. Environment variables are much cheaper to retrieve.

I think this particular value is fine to use from the environment because:
1. It does not change readily (or ever), so we would not need the ability to update it without a restart as a Variable.
2. It is [already defined as an environment variable](https://github.com/WordPress/openverse-catalog/blob/f42555b19ea2488feebe7ea01836e652a14158d8/env.template#L104-L103).
3. It is [used directly elsewhere](https://github.com/WordPress/openverse-catalog/blob/f654446ae34bb2351b5e52456404efa8c6002812/openverse_catalog/dags/providers/provider_dag_factory.py#L90) so changing this value would require a deployment/service restart anyway.

I hope that helps, let me know if you think it would be useful to codify some of this into an FAQ or as part of our README/contributing guide.

Reply:
Honestly, I'm pretty sure I was just copying what I had seen elsewhere, but I'm super grateful for the explanation here, @AetherUnbound !